static/app/utils/profiling/guards/profile.tsx

```diff
@@ -7,6 +7,11 @@ export function isSchema(input: any): input is Profiling.Schema {
     'shared' in input
   );
 }
+
+export function isNodeProfile(profile: any): profile is [Profiling.NodeProfile, {}] {
+  return Array.isArray(profile) && profile.length === 2 && isSampledProfile(profile[0]);
```

Comment:
Yes correct. For that second part, I'd need to check that it's an empty object. I think it's fine because it will only kick in if there's only 2 samples (which should never be the case for typescript profiles)